test/integration/testpaths/testpath_hypervisor_limit.py

```diff
@@ -0,0 +1,191 @@
+
+    def tearDown(self):
+        try:
+
+            cmd = updateHypervisorCapabilities.updateHypervisorCapabilitiesCmd()
+            cmd.id = self.hostCapId
+            cmd.maxguestslimit = self.originalLimit
+            self.apiclient.updateHypervisorCapabilities(cmd)
+
+            cleanup_resources(self.apiclient, self.cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+        return
+
+    @attr(tags=["advanced", "basic"], required_hardware="false")
+    def test_check_hypervisor_max_limit_effect(self):
+        """ Test hypervisor max limits effect
+
+        # 1. Read maxguestslimit global setting and
+                modify it to 4
+        # 2. Deploy a VM
+        # 2. Try to deploy another vm
+        # 3. Verify that second VM 
+                deployment fails (2 SSVMs 1 VR VM and 1 deployed VM)
+        """
+        qresultset = self.dbclient.execute(
+                "select hypervisor_version from host where uuid='%s'" % self.host.id)
+
+        event_validation_result = validateList(qresultset)
+        self.assertEqual(
+                event_validation_result[0],
+                PASS,
+                "event list validation failed due to %s" %
+                event_validation_result[2])
+         
+        # Step 1
+        cmdList = listHypervisorCapabilities.listHypervisorCapabilitiesCmd()
+        cmdList.hypervisor = self.hypervisor
+        config = self.apiclient.listHypervisorCapabilities(cmdList)
+
+        for host in config:
+            if host.hypervisorversion == qresultset[0][0]:
+                self.hostCapId = host.id
+                self.originalLimit = host.maxguestslimit
+                break
+        else:
+            self.skipTest("No hypervisor capabilities found for %s \
+                    with version %s" % (self.hypervisor, qresultset[0][0]))
+
+        cmdUpdate = updateHypervisorCapabilities.\
+                updateHypervisorCapabilitiesCmd()
+        cmdUpdate.id = self.hostCapId
+        cmdUpdate.maxguestslimit = "4"
+        self.apiclient.updateHypervisorCapabilities(cmdUpdate)
+
+        # Step 2
+        vm = VirtualMachine.create(
+            self.userapiclient,
+            self.testdata["small"],
+            templateid=self.template.id,
+            accountid=self.account.name,
+            domainid=self.account.domainid,
+            serviceofferingid=self.service_offering.id,
+            zoneid=self.zone.id,
+        )
+
+        self.cleanup.append(vm)
+        # Step 3
+        with self.assertRaises(Exception):
+            VirtualMachine.create(
+                self.userapiclient,
+                self.testdata["small"],
+                templateid=self.template.id,
+                accountid=self.account.name,
+                domainid=self.account.domainid,
+                serviceofferingid=self.service_offering.id,
+                zoneid=self.zone.id,
+            )
+
```

Comment:
If there are more than one host in a cluster, how can you make sure that all the vm deployments will go to the same host to test this functionality? IMO , it would be better to tag the host and try to deploy vms on it.